dlib/dnn/trainer.h

```diff
@@ -1057,34 +1081,39 @@ namespace dlib
             if (!std::ifstream(newest_syncfile(), std::ios::binary))
                 return false;
 
-            for (auto x : previous_loss_values)
+            // if we haven't seen much data yet then just say false.
+            if (gradient_updates_since_last_sync < 30)
+                return false;
```

Comment:
Move this after the NaN check.  Definitely want to reload when you hit a NaN, even if there are fewer than 30 updates.

Reply:
Good catch, fixed.